lib/components/form/user-settings.js

```diff
@@ -1,308 +1,424 @@
-import moment from 'moment'
-import coreUtils from '@opentripplanner/core-utils'
-import React, { Component } from 'react'
+/* eslint-disable @typescript-eslint/no-use-before-define */
+/* eslint-disable react/prop-types */
 import { Button } from 'react-bootstrap'
-import { FormattedMessage, injectIntl } from 'react-intl'
 import { connect } from 'react-redux'
+import { FormattedMessage, injectIntl, useIntl } from 'react-intl'
+import coreUtils from '@opentripplanner/core-utils'
+import moment from 'moment'
+import React, { Component, useCallback } from 'react'
+
+import * as apiActions from '../../actions/api'
+import * as formActions from '../../actions/form'
+import * as uiActions from '../../actions/ui'
+import * as userActions from '../../actions/user'
+import { getFormattedPlaces } from '../../util/i18n'
+import {
+  getPlaceDetail,
+  getPlaceMainText,
+  isHome,
+  isOtpMiddleware,
+  isWork
+} from '../../util/user'
+import { LinkWithQuery } from '../form/connected-links'
+import { StyledMainPanelPlace } from '../user/places/styled'
+import PlaceShortcut from '../user/places/place-shortcut'
 
-import { forgetSearch, toggleTracking } from '../../actions/api'
-import { setQueryParam } from '../../actions/form'
-import { forgetPlace, forgetStop, setLocation } from '../../actions/map'
-import { setViewedStop } from '../../actions/ui'
-import Icon from '../util/icon'
+import { UnpaddedList } from './styled'
 
-const { formatStoredPlaceName, getDetailText, matchLatLon } = coreUtils.map
-const { summarizeQuery } = coreUtils.query
+const { matchLatLon } = coreUtils.map
+const { hasTransit, toSentenceCase } = coreUtils.itinerary
+/**
+ * Version of summarizeQuery and helper function that supports i18n.
+ * FIXME: replace when the original does support i18n.
```

Comment:
Starting to feel like this PR is cursed, but do you want to block until https://github.com/opentripplanner/otp-ui/pull/330 is merged?

Reply:
I think they can be done independently.